weight-meter/src/lib.rs

```diff
@@ -0,0 +1,21 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::weights::Weight;
+
+struct Meter {
+	used_weight: Weight,
+	// Deep gets incremented when entering call or a sub-call
+	// This is used to avoid miscalculation during sub-calls
+	deep: usize,
```

Comment:
```suggestion
	depth: usize,
```

Reply:
why it is usize?

Reply:
No reason at all, u8 works as well. This is going to be 1 most of the time. `usize` was the easiest choice

Reply:
Should avoid usize because the size is not fixed. So could go u32 here.